libs/utils/test.py

```diff
@@ -100,4 +102,24 @@ def _experimentsFinalize(cls):
         Code executed after running the experiments
         """
 
+@wrapt.decorator
+def experiment_test(wrapped_test, instance, args, kwargs):
+    """
+    Convert a LisaTest test method to be automatically called for each experiment
+
+    The method will be passed the experiment object and a list of the names of
+    tasks that were run as the experiment's workload.
+    """
+    for experiment in instance.executor.experiments:
+        tasks = experiment.wload.tasks.keys()
+        try:
+            wrapped_test(experiment, tasks, *args, **kwargs)
+        except AssertionError as e:
+            trace_relpath = os.path.join(experiment.out_dir, "trace.dat")
+            e.args += ("Check trace file: " + os.path.abspath(trace_relpath),)
```

Comment:
This produces failure messages that look like this (copy-pasted from my terminal)

```
======================================================================
FAIL: Wake Migration: LITTLE -> BIG: 1
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/vagrant/lisa/libs/utils/test.py", line 116, in experiment_test
    wrapped_test(experiment, tasks, *args, **kwargs)
  File "/vagrant/tests/eas/acceptance.py", line 381, in test_little_big_switch1
    self._assert_switch(experiment, "big", 1)
  File "/vagrant/tests/eas/acceptance.py", line 376, in _assert_switch
    msg=msg)
AssertionError: ('Not all tasks wake-migrated to big cores in the expected window: (5.3126789999660105, 6.3126789999660105)', 'Check trace file: /vagrant/results/20161003_112951/rtapp::wake_migration/1/trac)

----------------------------------------------------------------------
Ran 5 tests in 47.696s
```

Note that in this case, where the original failure message is long, the trace file path is truncated by the width of my terminal; in order to see it I have to scroll to the right :disappointed: . 

The previous version of this logic didn't actually work; it prevented the test failures from being recognised by `nose` (it swallowed the `AssertionError`). When I fixed that mistake, the "Check trace file" message was no longer reported next to the test failure. So this isn't ideal.